src/test/java/uk/gov/hmcts/befta/dse/ccd/definition/converter/EnvironmentURLUtilsTest.java

```diff
@@ -0,0 +1,243 @@
+            = "/case_type/fe-functional-test/mid_event_dynamic_list";
+
+    @BeforeEach
+    void setup() throws IOException {
+        caseEventJson = objectMapper.readTree(
+                String.format(CASE_EVENT_CONTENT_TEMPLATE,
+                        NOC_REQUEST_AUTO_APPROVAL_SUBMIT_EVENT_HOST, NOC_REQUEST_AUTO_APPROVAL_SUBMIT_EVENT_PATH,
+                        APPLY_NOC_DECISION_SUBMIT_EVENT_HOST, APPLY_NOC_DECISION_SUBMIT_EVENT_PATH)
+        );
+    }
+
+    @Test
+    void unsupportedSheetNamesShouldNotModifyURLs() throws JsonProcessingException, MalformedURLException {
+        JsonNode nullJsonNode = objectMapper.nullNode();
+
+        JsonNode modifiedJsonNode = EnvironmentURLUtils.updateCallBackURLs(nullJsonNode, "AFileNameThatIsNotCaseEvent");
+        assertEquals(nullJsonNode, modifiedJsonNode);
+    }
+
+    @Test
+    @SetEnvironmentVariable(key = EnvironmentURLUtils.TEST_STUB_SERVICE_BASE_URL, value = LOCALHOST_URL)
+    void caseEventToFieldsSheetModifyTestStubURL() throws JsonProcessingException, MalformedURLException {
+        JsonNode caseEventToFieldsNode = objectMapper.readTree(String.format(CASE_EVENT_TO_FIELDS_TEMPLATE,
+                CREATE_CASE_CALLBACK_MID_EVENT_HOST,
+                CREATE_CASE_CALLBACK_MID_EVENT_PATH));
+
+        JsonNode modifiedJsonNode = EnvironmentURLUtils.updateCallBackURLs(caseEventToFieldsNode, "CaseEventToFields");
+
+        // check JSON has been modified
+        assertNotEquals(caseEventToFieldsNode, modifiedJsonNode);
+
+        // check placeholders are not still present
+        assertFalse(modifiedJsonNode.toString().contains(CREATE_CASE_CALLBACK_MID_EVENT_HOST));
+
+        // check URLs have been substituted with correct values
+        assertTrue(modifiedJsonNode.toString().contains(LOCALHOST_URL+ CREATE_CASE_CALLBACK_MID_EVENT_PATH));
+    }
+
+    @Test
+    void caseEventToFieldsUpdatedWithDefaultValuesWheNoEnvironmentVariablesSet()
+            throws JsonProcessingException, MalformedURLException {
+        JsonNode caseEventToFieldsNode = objectMapper.readTree(String.format(CASE_EVENT_TO_FIELDS_TEMPLATE,
+                CREATE_CASE_CALLBACK_MID_EVENT_HOST,
+                CREATE_CASE_CALLBACK_MID_EVENT_PATH));
+
+        JsonNode modifiedJsonNode = EnvironmentURLUtils.updateCallBackURLs(caseEventToFieldsNode, "CaseEventToFields");
+
+        // check JSON has been modified
+        assertNotEquals(caseEventToFieldsNode, modifiedJsonNode);
+
+        // check placeholders are not still present
+        assertFalse(modifiedJsonNode.toString().contains(CREATE_CASE_CALLBACK_MID_EVENT_HOST));
+
+        // check URLs have been substituted with correct values
+        assertTrue(modifiedJsonNode.toString()
+                .contains(CREATE_CASE_CALLBACK_MID_EVENT_DEFAULT_HOST + CREATE_CASE_CALLBACK_MID_EVENT_PATH));
+    }
+
+    @Test
+    void caseEventUrlsUpdatedWithDefaultValuesWheNoEnvironmentVariablesSet() throws IOException {
```

Comment:
Typo in name `When`

Reply:
Resolved in https://github.com/hmcts/befta-fw/pull/123/commits/f7fbff8c44e377ce7f637f1419a516c10c184c61